src/app/Library/CrudPanel/CrudField.php

```diff
@@ -430,8 +435,14 @@ public function dd()
      */
     public function __call($method, $parameters)
     {
-        $this->setAttributeValue($method, $parameters[0]);
+        $macro = static::$macros[$method] ?? null;
 
-        return $this->save();
+        if (! $macro) {
+            $this->setAttributeValue($method, $parameters[0]);
+
+            return $this->save();
+        }
+
+        return call_user_func_array($macro->bindTo($this, static::class), $parameters);
```

Comment:
We have the `Macroable` trait on `CrudField`. So... why not use its methods, instead of doing plain PHP? I believe this will do the same, but cleaner:

```php
        if (static::hasMacro($method)) {
            return $this->macroCall($method, $parameters);
        }

        $this->setAttributeValue($method, $parameters[0]);

        return $this->save();
```

What it does different:
- it uses `hasMacro()` and `callMacro()` instead of PHP which is more difficult to read and understand;
- it changes the conditional from negative to positive;
- it keeps the same default behaviour as before; the default behaviour is to set the attribute - this change just adds an "unless" before that;

Reply:
That way, we keep the "return early" mentality:

```php
public function __call() {
    if ($unlikely_event) {
        return $early;
    }
    
    return $what_is_the_most_likely_scenario;
}
```

This makes a lot more sense to me than the current:

```php
public function __call() {
    $unlikely_condition = $maybe ?? false;

    if ($not_unlikely_condition) {
        return $most_likely_scenario;
    }
    
    return $unlikely_event;
}
```

You know what I mean Pedro? 

**Rule of thumb:** if your conditional starts with a negative... there's probably a simpler way to phrase that condition.